src/simpleNbox.cpp

```diff
@@ -595,6 +602,10 @@ unitval SimpleNbox::getData(const std::string& varName,
     } else if( varNameParsed == D_LUC_EMISSIONS ) {
         H_ASSERT( date != Core::undefinedIndex(), "Date required for luc emissions" );
         returnval = lucEmissions.get( date );
+    } else if( varNameParsed == D_CA_CONSTRAIN ) {
+        H_ASSERT( date != Core::undefinedIndex(), "Date required for atmospheric CO2 constraint" );
+        H_ASSERT( Ca_constrain.exists(date), "No CO2 constraint for selected date" );
```

Comment:
> Is there a way we could return a sentinel value that the R interface could translate to an NA value?

Is `NaN` OK? Or, more formally, how important is it that we maintain the distinction between `NaN` (the result of an illegal math operation, like dividing by zero) and `NA` (a true missing value)? Blurring the distinction makes this easy -- it's trivial to set `std::NAN` as the return value, and R has native support for those. Preserving the distinction would be more challenging (though could ultimately be rewarding down the line), since it would require more substantial modifications of `sendmessage` and/or `unitval`.

My proposal would be to stick with `NaN` for now and discuss implementing a true missing value class later on.

Reply:
Actually, I think I found a workaround (4d348ba): If Rcpp is available, we can use its `NA_REAL`, which R (and compatible Rcpp code) understands to be `NA` but other C++ code will treat as `NAN`. At least for the R package, this lets us have our cake and eat it too, without (AFAICT) any major drawbacks.

Reply:
Sounds great.  I was going to say return a `NaN` and have the R code change it to `NA`, but this is even better.